tgui/packages/tgui/interfaces/ChemHeater.js

```diff
@@ -0,0 +1,130 @@
+        <Section title="Temperature Control">
+
+          <Stack align="center">
+            <Stack.Item>
+              <Knob
+                animated
+                size={2}
+                value={targetTemperature}
+                minValue={0}
+                maxValue={1000}
+                format={value => value + " K"}
+                onDrag={(e, value) => act('adjustTemp', { temperature: value })}
+              />
+            </Stack.Item>
+            <Stack.Item grow overflow="hidden" py={1}>
+              <Box
+                className="TemperatureNumber"
+                nowrap
+                p={1}
+                fontSize={1.5}
+                color={getTemperatureColor(targetTemperature)}
+                backgroundColor="black">
+                <Box fontSize={1}>Target</Box>
+                <Icon name={getTemperatureIcon(targetTemperature)} pr={0.5} />
+                <AnimatedNumber value={targetTemperature} /> K
+              </Box>
+            </Stack.Item>
+            <Stack.Item basis={9.6} align="center">
+              <Button
+                icon="power-off"
+                disabled={!(containerData && containerData.totalVolume)}
+                color={isActive ? "red" : "green"}
+                fluid
+                height="100%"
+                fontSize={1.25}
+                textAlign="center"
+                onClick={() => act(isActive ? 'stop' : 'start')}>
+                {isActive ? "Deactivate" : "Activate"}
+              </Button>
+            </Stack.Item>
+          </Stack>
+
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const ChemDisplay = (props, context) => {
+  const { act } = useBackend(context);
+  const { active = false, targetTemperature = 273.15 } = props;
+  const container = props.container || NoContainer;
+  const working = active && !container.fake;
+  const { temperature, totalVolume } = container;
+
+  return (
+    <Section
+      title={
+        <Flex inline nowrap>
+          <Flex.Item grow
+            overflow="hidden"
+            style={{
+              "text-overflow": "ellipsis",
+              "text-transform": "capitalize",
+            }}>
+            {container.name}
+          </Flex.Item>
+          <Flex.Item px={4.25} /> {/* button overlap prevention */}
+        </Flex>
+      }
+      buttons={
+        <Button
+          icon="eject"
+          disabled={!props.container}
+          onClick={() => act('eject')}>
+          Eject
+        </Button>
+      }>
+      <ReagentGraph container={container} />
+      <ReagentList container={container} />
+      <Box className={"TemperatureBox " + (working && `TemperatureBox__${getTemperatureChange(temperature, targetTemperature)}`)}>
+        {!totalVolume || (
+          <Box
+            fontSize={2}
+            color={getTemperatureColor(temperature)}
+            className={"TemperatureNumber"}>
```

Comment:
```
            className="TemperatureNumber">
```
But also change to namespace or make this into a common component, as previous comment.

Reply:
Clarifying, this should be a component on its own, or `ChemHeater__TemperatureNumber`. I didn't spell it out for simplicity, but realized that may have caused confusion.

Reply:
This was a mistake, I forgot to change it to `ChemHeater_TemperatureNumber` and didn't notice.